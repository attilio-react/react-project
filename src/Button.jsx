import React from 'react';

class Button extends React.PureComponent {
  render () {
    const {selected} = this.props
    return <input type="button" value={this.props.caption} style={{textDecoration: selected ? 'underline' : 'none'}}></input>;
  }
}


export {Button}


