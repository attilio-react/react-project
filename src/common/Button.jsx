import React from 'react';

class Button extends React.PureComponent {
  render () {
    const {selected, onClick, caption} = this.props
    return <input type="button" onClick={onClick}  value={caption} style={{textDecoration: selected ? 'underline' : 'none'}}></input>;
  }
}


export {Button}


