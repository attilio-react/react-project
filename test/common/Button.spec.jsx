import * as React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';


import {Button} from '|common/Button.jsx';

describe('Button rendering', () => {
  it('Should render selected', () => {
    const component = renderer.create(<Button value='hello' selected='true' />),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('Should render not selected', () => {
    const component = renderer.create(<Button value='hello' />),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Button behaviour', () => {
  it('Should call callback when clicked', () => {
    const onClick = jest.fn(),
      button = mount(<Button value='hello' onClick={onClick} />),
      nativeButton = button.find('input').first();
    nativeButton.simulate('click');
    expect(onClick).toBeCalled()
  });
});

