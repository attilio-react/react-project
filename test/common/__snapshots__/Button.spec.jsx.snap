// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button rendering Should render not selected 1`] = `
<input
  style={
    Object {
      "textDecoration": "none",
    }
  }
  type="button"
/>
`;

exports[`Button rendering Should render selected 1`] = `
<input
  style={
    Object {
      "textDecoration": "underline",
    }
  }
  type="button"
/>
`;
